src/Symfony/Component/Config/Definition/Builder/VariableNodeDefinition.php

```diff
@@ -54,7 +54,9 @@ protected function createNode()
         $node->addEquivalentValue(true, $this->trueEquivalent);
         $node->addEquivalentValue(false, $this->falseEquivalent);
         $node->setRequired($this->required);
-        $node->setDeprecated($this->deprecationMessage);
+        if ([] !== $this->deprecation) {
+            $node->setDeprecated(...array_values($this->deprecation));
```

Comment:
I think we should rather read the keys explicitly than relying on the order being the same than arguments (which might be easy to break by mistake).

Reply:
done